### PR TITLE
Structure for updated BidderActions container

### PR DESCRIPTION
**Pull request by l-ohman**
- BidderActions is now based on the subdomain
- New "Your Team's Bid" component
- New "Your Supporters" component
- Container for captains to toggle between roster/supporters

Updated BidderActions container for NADCL

### CODE REVIEW

File: frontend/pages/auction/[auction].tsx [hunk 1]
```diff
@@ -77,6 +78,14 @@ export default function Draft() {
...
   );
 
+  // add subdomain to global store so all components have easy access
+  const subdomain = useGlobalStore((state) => state.subdomain);
+  const setSubdomain = useGlobalStore((state) => state.setSubdomain);
+  if (!subdomain && typeof window !== "undefined") {
+    const updatedSubdomain = parseSubdomainIfExists(window.location.hostname);
+    setSubdomain(updatedSubdomain);
+  }
+
```
Comment (cadenforrest) [inline frontend/pages/auction/[auction].tsx:88]: What happens if someone visits `playerbase.fun/auction/4` (no subdomain) where auction 4 is a supporter auction?

File: frontend/components/auction/BidderActions/_BidderActions.tsx [hunk 2]
```diff
@@ -1,7 +1,19 @@
...
+  YourTeamsBid,
+  RosterSupportersSelector,
+} from "./";
+import { useGlobalStore } from "@/store";
 
 export default function BidderActions() {
+  const subdomain = useGlobalStore((state) => state.subdomain);
+  if (subdomain?.toLowerCase() === "nadcl") return <BidderActionsNADCL />;
+  return <BidderActionsDefault />;
+}
```
Comment (cadenforrest) [inline frontend/components/auction/BidderActions/_BidderActions.tsx:14]: Ditto here - could we not look at auction.type instead? 

```typescript
const { data: auction }  = useSWR(...);
if (auction.type === "SUPPORTER") return <BidderActionsNADCL />;
return <BidderActionsDefault />;
```  

As currently written, we'll have unexpected behavior if we're accessing the auction from no subdomain

**Review by cadenforrest — CHANGES_REQUESTED** [review]
It looks like there are only a couple of places where we reference the subdomain here, which may cause issues for supporter auctions if there is no subdomain present - for this PR, can we replace that with references to the auction type? 

I think having the subdomain in a global store like this is a nice pattern and will help with a lot of the things we're trying to do

File: frontend/pages/auction/[auction].tsx (hunk 1 above)
Comment (l-ohman) [reply frontend/pages/auction/[auction].tsx]: The "subdomain" in the global store will simply be set to `null`. There's default behavior for everything that relies on subdomain, so it should handle things properly in that specific case.

File: frontend/pages/auction/[auction].tsx [hunk 3]
```diff
@@ -77,6 +78,22 @@ export default function Draft() {
...
   );
 
+  // these should probably only run when the auction is first loaded...
+  // add subdomain to global store
+  const subdomain = useGlobalStore((state) => state.subdomain);
+  const setSubdomain = useGlobalStore((state) => state.setSubdomain);
+  if (!subdomain && typeof window !== "undefined") {
+    const updatedSubdomain = parseSubdomainIfExists(window.location.hostname);
+    setSubdomain(updatedSubdomain);
+  }
```
Comment (cadenforrest) [inline frontend/pages/auction/[auction].tsx:88]: You could even rip this out into the app layout or something if you wanted to get real spicy with it

**Review by cadenforrest — APPROVED** [review]

**Pull request merged**